### PR TITLE
Outlet boundary condition

### PR DESCRIPTION
**Pull request by blaisb**
# Description of the problem

- Currently only the do-nothing outlet boundary condition was implemented. This boundary condition does not prevent backflow inside of the domain and this can be problematic.

# Description of the solution

- Implement a realistic outlet boundary condition that prevents backflow

# How Has This Been Tested?

- The cylinder test has been modified to use the new outlet boundary condition. it does not change anything but at least it shows that the implementation is stable. Not sure if it is necessary to add another test for this.

# Documentation

- I changed the documentation to discuss this boundary condition and I took the opportunity to add some information about the weak function dirichlet bc.

# Comments

- Eventually i'll try to come up with a test where this pops up, but I think it could be done in another PR if we use this feature a lot in the long term.

### CODE REVIEW

**Review by Luckabarbeau — COMMENTED** [review]
A small comment on the beta term. I would add a test with the vans for weak function ,pressure and outlet BC

**Review by tonielgeitani — APPROVED** [review]

File: doc/source/parameters/cfd/boundary_conditions_cfd.rst [hunk 1]
```diff
@@ -50,6 +64,8 @@ This subsection defines the boundary conditions associated with fluid dynamics p
 
 * ``time dependent`` specifies if a  boundary condition is time dependent (``true``) or steady (``false```). By default, this parameter is set to ``false``. This is there to improve the computational efficiency for transient cases in which the boundary conditions do not change. 
 
+* ``beta`` is a penalization paramter that is used for both the ``outlet`` and the ``function weak`` boundary conditions. For the outlet boundary conditions, ``beta`` should be close to unity whereas values of ``beta`` for 10 or a 100 can be appropriate for the ``function weak`` boundary condition.
+
```
[the remark this answers is not in the document]
Comment (blaisb) [reply doc/source/parameters/cfd/boundary_conditions_cfd.rst:68]: Done
